packages/vite/src/node/plugins/resolve.ts

```diff
@@ -803,11 +803,11 @@ export function tryNodeResolve(
     : OPTIMIZABLE_ENTRY_RE.test(resolved)
 
   let exclude = depsOptimizer?.options.exclude
-  let include = depsOptimizer?.options.exclude
+  let include = depsOptimizer?.options.include
   if (options.ssrOptimizeCheck) {
     // we don't have the depsOptimizer
     exclude = options.ssrConfig?.optimizeDeps?.exclude
-    include = options.ssrConfig?.optimizeDeps?.exclude
+    include = options.ssrConfig?.optimizeDeps?.include
```

Comment:
I guess this was incorrect. But I'm not sure how it effects. I think we should have a test that covers this. (I don't understand what test case we should have though.)
Maybe I should split this part into another PR and merge that first?

Reply:
I think it's probably fine to do a quick fix for this and we can add tests later when we solidify SSR optimization.